### PR TITLE
fix: sort promotions with running and pending first

### PR DESCRIPTION
**Pull request by rbreeze**
Fixes https://github.com/akuity/kargo/issues/1037

@jessesuen In the issue above, you mentioned adding a special section in the UI for currently running promotions. This seems like a separate issue to me, but I'm happy to keep the above issue open if you prefer to track it there

### CODE REVIEW

File: ui/src/features/stage/promotions.test.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  pending11,
+  pending9,
+  pending4,
+  pending2,
+  unknown14,
+  errored13,
+  succeeded12,
+  unknown7,
+  failed6,
+  succeeded5
```
Comment (rbreeze) [inline ui/src/features/stage/promotions.test.ts:68]: Running promotions come first, sorted by timestamp. Then pending promotions, also sorted by timestamp, and finally everything else sorted by timestamp.

File: ui/src/features/stage/promotions.tsx [hunk 2]
```diff
@@ -95,7 +90,11 @@ export const Promotions = () => {
         switch (promotion.status?.phase) {
           case 'Succeeded':
             return (
-              <Popover content={promotion.status?.message} title={promotion.status?.phase} placement='right'>
+              <Popover
+                content={promotion.status?.message}
+                title={promotion.status?.phase}
+                placement='right'
+              >
```
Comment (rbreeze) [inline ui/src/features/stage/promotions.tsx:97]: Nothing changed here, just ran Prettier

File: ui/src/features/stage/promotions.tsx [hunk 3]
```diff
@@ -106,7 +105,11 @@ export const Promotions = () => {
           case 'Failed':
           case 'Errored':
             return (
-              <Popover content={promotion.status?.message} title={promotion.status?.phase} placement='right'>
+              <Popover
+                content={promotion.status?.message}
+                title={promotion.status?.phase}
+                placement='right'
+              >
```
Comment (rbreeze) [inline ui/src/features/stage/promotions.tsx:112]: Same here, Prettier fix

**Review by jessesuen — APPROVED** [review]
We'll eventually want the running promotion displayed prominently, e.g. with details to the PR, etc... Also there can only be on running PR at a time so we can reserve special place in the UI for it.

**Pull request merged**